scripts/extract_business_rules.py

```
@@ -0,0 +1,201 @@
+                parts.append(str(val.value))
+            elif isinstance(val, ast.FormattedValue):
+                parts.append(f"{{{ast.unparse(val.value)}}}")
+        return "".join(parts)
+    elif isinstance(node, ast.Call) and isinstance(node.func, ast.Attribute) and node.func.attr == 'format':
+        return ast.unparse(node)
+    return ast.unparse(node)
+
+def process_file(filepath):
+    with open(filepath, 'r') as f:
+        source = f.read()
+    
```

Pylint (4.0.5)

[convention] Trailing whitespace

(C0303)